worker/src/worker.rs

```diff
@@ -457,16 +442,25 @@ impl WorkerToWorker for WorkerReceiverHandler {
                 .send(batch)
                 .await
                 .map_err(|_| DagError::ShuttingDown),
-
-            WorkerMessage::BatchRequest(missing, requestor) => self
-                .tx_worker_helper
-                .send((missing, requestor))
-                .await
-                .map_err(|_| DagError::ShuttingDown),
         }
         .map(|_| anemo::Response::new(()))
         .map_err(|e| anemo::rpc::Status::internal(e.to_string()))
     }
+    async fn request_batches(
+        &self,
+        request: anemo::Request<types::WorkerBatchRequest>,
+    ) -> Result<anemo::Response<types::WorkerBatchResponse>, anemo::rpc::Status> {
+        let message = request.into_body();
+        let batches: Vec<Batch> = self
+            .store
+            .read_all(message.digests)
+            .await
+            .map_err(|e| anemo::rpc::Status::from_error(Box::new(e)))?
+            .into_iter()
+            .flatten()
+            .collect();
+        Ok(anemo::Response::new(types::WorkerBatchResponse { batches }))
```

Comment:
If we expect batches to be large then it may be better to have this rpc only handle a single batch request and instead request a Vec of batches by initiating N requests.

Reply:
I could implement a max request size of N batches, which would also help towards addressing the issue above? Can any of you advise on a good value for N? I don't have a sense of how big these are, either.

Reply:
@huitseeker

Reply:
I put a TODO for this next to the one about defending against malicious workers using excessive resources

Reply:
Do we have an idea of the upper-limit on size of a batch in terms of bytes?